aten/src/ATen/mps/MPSDevice.mm

```diff
@@ -0,0 +1,41 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/mps/MPSDevice.h>
+
+namespace at {
+namespace mps {
+
+MPSDevice* MPSDevice::_device = nullptr;
+
+MPSDevice* MPSDevice::getInstance() {
+  if (_device == nullptr) {
+    _device = new MPSDevice();
```

Comment:
intentional leak, right? This doesn't look thread safe

Reply:
Good catch. Fix coming up..

Reply:
This is fixed now.